### ISSUE TITLE
[S026] Assemble the full GPT-2 forward pass and tied output head

### ISSUE THREAD

**Issue by 25d-228** [25d-228/gpt-2#28]
**Prerequisites:** S009, S012, S015, S024–S025; #1 applies. **Timebox:** 40–50 minutes.

## Learn and implement
Create `include/gpt2.h`, `src/gpt2.c`, and `tests/test_gpt2_forward.c`. Wire token+position embeddings → L transformer blocks in order → final LayerNorm → logits `[B,T,V]` using the token embedding matrix as `[V,C]` output weights and null bias. Validate requests and IDs before computation. The forward call must not modify parameters. Use fixed test parameters for now; initialization is S030.

## Explicit tests
All-zero parameters yield zero logits. For `V=3,C=2,H=1,L=1,P=2,T=1`, use token rows `[[1,0],[0,1],[1,1]]`, zero positions, unit norm scales, zero norm biases, and zero branch output projections/biases. Input ID 0 gives logits `[a,-a,0]`, where `a=0.5/sqrt(0.25+1e-5)`. Compute a independently in the test. Changing only unused token row 2 to `[2,1]` must change its logit to a without changing the hidden state: this tests weight tying. Also test two layers, two batch rows, and unchanged parameter bytes.
```sh
make test TEST=gpt2_forward
make sanitize TEST=gpt2_forward
make test
```
All pass with #1 tolerances. Temporarily skip final LayerNorm and demonstrate failure.

## Done / teach-back
Trace the complete shape pipeline and explain why the output is unnormalized logits rather than generated text. No loss, sampling, or backward.
